autochain/agent/structs.py

```diff
@@ -55,30 +54,94 @@ def format_output(self) -> Dict[str, Any]:
 
 
 class AgentOutputParser(BaseModel):
```

Comment:
i think there is a way to simplify this logics inside of this function
first json load, if run into exception, fix and generate, format into message for `load_json_output`, decrease retry, call load_json_output again, 
if no exception, just return